vars/checkResultsPublish.groovy

```diff
@@ -0,0 +1,147 @@
+        def doArchive = parameters.get('archive', false)
+        // JAVA
+        def pmd = parameters.get('pmd', false)
+        def cpd = parameters.get('cpd', false)
+        def findbugs = parameters.get('findbugs', false)
+        def checkstyle = parameters.get('checkstyle', false)
+        // JAVA SCRIPT
+        def eslint = parameters.get('eslint', false)
+        // PYTHON
+        def pylint = parameters.get('pylint', false)
+
+        // report TODOs
+        reportTasks(tasks, '**/*.java')
+        // report PMD
+        report('PmdPublisher', pmd, '**/target/pmd.xml', doArchive)
+        // report CPD
+        report('DryPublisher', cpd, '**/target/cpd.xml', doArchive)
+        // report Findbugs
+        report('FindBugsPublisher', findbugs, '**/target/findbugsXml.xml, **/target/findbugs.xml', doArchive)
+        // report Checkstyle
+        report('CheckStylePublisher', checkstyle, '**/target/checkstyle-result.xml', doArchive)
+        // report ESLint
+        reportWarnings('JSLint', eslint, '**/eslint.xml', doArchive)
+        // report PyLint
+        reportWarnings('PyLint', pylint, '**/pylint.log', doArchive)
+
+        // aggregate results
+        aggregate(aggregation)
+    }
+}
+
+def aggregate(settings){
+    if (!Boolean.FALSE.equals(settings)) {
+        settings = asMap(settings)
+        def options = createCommonOptionsMap('AnalysisPublisher', settings)
+        // publish
+        step(options)
+    }
+}
+
+def report(stepName, settings, defaultPattern, doArchive){
+    // exit if set to FALSE
+    if(!Boolean.FALSE.equals(settings)){
+        settings = asMap(settings)
+        def pattern = settings.get('pattern', defaultPattern)
+        def options = createCommonOptionsMap(stepName, settings)
+        options.put('pattern', pattern)
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive', 'true'), pattern, true)
+    }
+}
+
+def reportWarnings(parserName, settings, defaultPattern, doArchive){
+    // exit if set to FALSE
+    if(!Boolean.FALSE.equals(settings)){
+        settings = asMap(settings)
+        def pattern = settings.get('pattern', defaultPattern)
+        def options = createCommonOptionsMap('WarningsPublisher', settings)
+        options.put('parserConfigurations', [[
+            parserName: parserName,
+            pattern: pattern
+        ]])
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive', 'true'), pattern, true)
+    }
+}
+
+def reportTasks(settings, defaultPattern){
+    // exit if set to FALSE
+    if(!Boolean.FALSE.equals(settings)){
+        settings = asMap(settings)
+        def options = createCommonOptionsMap('TasksPublisher', settings)
+        options.put('pattern', settings.get('pattern', defaultPattern))
+        options.put('high', settings.get('high', 'FIXME'))
```

Comment:
I would suggest to move also this default configuration out of the step into the central default configuration yml